src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -261,11 +293,30 @@ public unsafe int Seek(ref Vector<byte> byte0Vector)
                             {
                                 following -= _vectorSpan;
                                 index += _vectorSpan;
+                                bytesScanned += _vectorSpan;
+
+                                if ((bytesScanned > limitBytes) ||
+                                    (!limitIterator.IsDefault && block == limitIterator.Block && index > limitIterator.Index))
```

Comment:
I think all these `!limitIterator.IsDefault` checks are unnecessary considering `block` will never be null.